Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/ScrollingObjectCollection.cs

```diff
@@ -0,0 +1,2198 @@
+            }
+            else
+            {
+                workingScrollerPos.x = initialScrollerPos.x = finalPos.x;
+            }
+
+            animatingToPosition = false;
+
+            if (callback != null)
+            {
+                callback?.Invoke();
+            }
+        }
+
+        /// <summary>
+        /// Checks to see if the engaged joint has released the scrollable list
+        /// </summary>
+        /// <param name="initialDirection">The plane normal direction.</param>
+        /// <param name="initialPosition">The point representing the plane's origin.</param>
+        /// <param name="pointToCompare">The point compared to the normal and origin.</param>
+        /// <param name="clippingObj">The object representing the maximum scrollable area.</param>
+        /// <param name="transformMatrix">The world Matrix for the scrollable area to be compared in.</param>
+        /// <param name="direction"><see cref="ScrollDirectionType"/> the list is scrolling in.</param>
+        /// <returns><see cref="true"/> if released.</returns>
+        private static bool DetectScrollRelease(Vector3 initialDirection, Vector3 initialPosition, Vector3 pointToCompare, Transform clippingObj = null, Matrix4x4? transformMatrix = null, ScrollDirectionType? direction = null)
+        {
+            //true if finger is on the other side (Z) of the initial contact point of the collection
+            if (pointToCompare.IsOtherSideOfPoint(initialDirection, initialPosition))
+            {
+                return true;
+            }
+
+            bool hasPassedBoundary = false;
+
+            if (clippingObj != null && transformMatrix != null && direction != null)
+            {
+                Matrix4x4 tMat = (Matrix4x4)transformMatrix;
+                Vector3 posToClip = tMat.MultiplyPoint3x4(pointToCompare) - clippingObj.localPosition;
+                Vector3 halfScale = clippingObj.localScale * 0.5f;
+
+                if (direction == ScrollDirectionType.UpAndDown)
+                {
+                    hasPassedBoundary = (posToClip.y > halfScale.y || posToClip.y < -halfScale.y) ? true : false;
+                }
+                else
+                {
+                    hasPassedBoundary = (posToClip.x > halfScale.x || posToClip.x < -halfScale.x) ? true : false;
+                }
+            }
+
+            return hasPassedBoundary;
+        }
+
+        /// <summary>
+        /// Grab all child renderers in each node from NodeList and add them to the ClippingBox
+        /// </summary>
+        private void AddAllItemsToClippingObject()
+        {
+            //Register all of the renderers to be clipped by the clippingBox
+            foreach (ObjectCollectionNode node in NodeList)
+            {
+                Renderer[] childRends = node.Transform.gameObject.transform.GetComponentsInChildren<Renderer>(true);
+                for (int i = 0; i < c
```

Comment:
Hold up, I thought you were doing touch detection in your Update() loop. What is that code doing then?